crates/matrix-sdk-ui/src/timeline/mod.rs

```diff
@@ -595,7 +606,13 @@ impl Timeline {
             }
         }
 
-        self.room().send_multiple_receipts(receipts).await
+        if !receipts.is_empty() {
+            self.room().send_multiple_receipts(receipts).await?;
+        } else if self.room().is_marked_unread() {
+            self.room().set_unread_flag(false).await?;
+        }
```

Comment:
Can we add `let room = self.room();` to avoid calling `self.room()` multiple time please?

Reply:
Done.